### DISCUSSION TITLE
Why don't I see the data that Ajax returns when I use the browser to view the source code？

### DISCUSSION THREAD

**Discussion by gh0sthx** [nuxt/nuxt#9482] [Questions (Nuxt 2)]
Does this mean that the data returned by Ajax cannot be retrieved by Baidu crawler?

This is the address of the NUTX project     **[https://juejin.cn](https://juejin.cn)**

**Comment by kissu** [comment]
If you disable JavaScript on this page, the actual content will not be loaded. A crawler will usually prefer static HTML content out of the box. It can understand and wait for the JS to kick in, but this is not the best in terms of performance/SEO.

When you check the source code (`ctrl + u`), you'll get what is static upon initial render, so no actual AJAX-fetched  content.

**Reply by gh0sthx** [reply]
use asyncData i can see AJAX-fetched content in the source code ，Right?

**Reply by kissu** [reply]
You're navigating when doing this, a client side thing. This is not what a crawler is doing. A crawler is essentially going to a webpage and looking what is the HTML there, he does not use JS so far. Or at least, he will not give you an awesome score if it *needs* to do so.
